### ISSUE TITLE
Questions about eta(η) and some understandings from myself wanted to be confirmed

### ISSUE THREAD

**Issue by ChanglongZheng** [LehengTHU/AdvInfoNCE#1]
Firstly thanks to your insightful paper about InfoNCE which helps us to use contrasive learning in recommendation more properly. Here are some questions( maybe kind of straightfoward, lol ):
### Questions about $\eta$

1. I noticed in the paper that $\epsilon$ ought to be a hyperparemeter so we can compute $\eta$ , in the paper you had set $\epsilon$ controlled by the number of adversarial training epochs, dose it refers to the parameter 'eta_epochs'? When we change 'eta_epochs' equals to we change $\epsilon$ and equals to we change $\eta$ ? Incidentally, I'm also confused by parameters 'eta_epochs' and 'adv_epochs'.
![image](https://github.com/LehengTHU/AdvInfoNCE/assets/58901253/148fc8f7-4f2c-4740-a143-502ae9bc3cd4)
![image](https://github.com/LehengTHU/AdvInfoNCE/assets/58901253/14a93ddd-3a76-40dd-b5e6-042f22ae371a)
2. $\eta$ also appeared in parse.py, but I didn't find the usages of it(maybe I lossed some infomation), according to the assumption above, the parameter 'eta' is not needed, right? 
![image](https://github.com/LehengTHU/AdvInfoNCE/assets/58901253/95c32ad6-2486-47cc-bbf4-0434b6e09511)

### Some undertandings wanted to be confirmed
1. Can I attribute the success of AdvInfoNCE to the success of distinguishing the 'true negative sample' and the 'false negative sample'? My thought is that a rough negative sampling method might bring noise into the contrasive loss, for we can't identify an item which hasn't be interactd a real negative item or false negative item, a fine-grained ranking criterion helps to reduce the noise caused this ambiguity.
2. As we know and you had mentioned in the paper that the hardness for a paticular $\delta_j$ can't be reached, so in the **Algorithm 1** does the inner max optimization is trying to compute $\delta_j$ ? (As a replacement of the real and inaccessible $\delta_j$)
![image](https://github.com/LehengTHU/AdvInfoNCE/assets/58901253/d110bf05-0d22-4cbd-9a46-afc9bf3d81be)
![image](https://github.com/LehengTHU/AdvInfoNCE/assets/58901253/a3bf6f43-6aef-4885-858f-ef67ca434778)
 
I feel a bit sorry for so many questions, I'm a rookie in the recommendation field so the questions might be immature. But I'm still looking forward to your reply, thanks :)

**Comment by LehengTHU** [comment]
Thanks for reading our paper and code so carefully!
1. For your first question about $\eta$. "change 'eta_epochs' equals to we change $\epsilon$ and equals to we change $\eta$ ?". Yes, your understanding is correct. 'eta_epochs' controls the strength of disturbance. Sorry for the confusion term 'adv_epochs'. This hyperparameter actually controls the training epochs for each adversarial training turn. In our setting, this hyperparameter is always set to 1.
2. The 'eta' you mentioned is actually not used. Sorry for this confusion. This is an early code that we forget to delete.
3. Your understanding about the fine-grained ranking criterion is also correct. False negative identification is the main reason of the success, especially when facing out-of-distribution testing datasets.
4. Your understanding about the algorithm is also correct. The inner max optimization is trying to compute $\delta_j$.
Hope my explanation helps. Feel free to contact me if there is still any confusion. I will update my code later and make it more clear when I'm available.

**Comment by ChanglongZheng** [comment]
I think all my questions have been solved! Thanks for your quick reply(👍), hoping to see your more insightful works :P
